akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -206,9 +225,10 @@ private[cluster] final class ClusterHeartbeatSender extends Actor with ActorLogg
 
   }
 
-  def heartbeatRsp(from: UniqueAddress): Unit = {
-    if (verboseHeartbeat) logDebug("Heartbeat response from [{}]", from.address)
-    state = state.heartbeatRsp(from)
+  def heartbeatRsp(response: HeartbeatRsp): Unit = {
+    // TODO: log response time and validate sequence nrs once serialisation of sendTime is released
```

Comment:
how will the validation of the sequence nrs work? we would have a Map of latest seen `HeartbeatRsp.sequenceNr` and verify that incoming seq nr is +1?

We are using a global counter for the outgoing messages so they may not start at 1 for a given address, but I think that is good enough. Otherwise we would have to use two maps.

Reply:
> we would have a Map of latest seen HeartbeatRsp.sequenceNr and verify that incoming seq nr is +1?

Yes I think that is good enough, with the first response assumed to be the starting seqNr